src/main/java/tahub/contacts/model/grade/GradingSystem.java

```diff
@@ -35,7 +41,17 @@ public double getGrade(String assessmentName) {
      * @param score the numerical score for the assessment (as a percentage)
      */
     public void addGrade(String assessmentName, double score) {
-        assessmentScores.put(assessmentName, score);
+        double weight = getExistingWeight(assessmentName);
+        grades.removeIf(g -> g.getAssessmentName().equals(assessmentName));
+        grades.add(new Grade(assessmentName, score, weight));
+    }
+
+    private double getExistingWeight(String assessmentName) {
+        return grades.stream()
+                .filter(g -> g.getAssessmentName().equals(assessmentName))
+                .findFirst()
+                .map(Grade::getWeight)
+                .orElse(0.0);
```

Comment:
Would be better to throw exception if assessmentName not found instead of silently returning 0

Reply:
okie will do and push